### PR TITLE
Fix horizontal scroll using trackpad gesture

### PR DESCRIPTION
**Pull request by EdnY1**
This allows horizontal scroll using the macbook's trackpad or any trackpad with the 2 fingers scroll gesture.

### CODE REVIEW

**Comment by tishion** [comment]
Need to confirm with you what problems you are solving?

File: src/details/QCefViewPrivate.cpp [hunk 1]
```diff
@@ -1036,18 +1036,22 @@ QCefViewPrivate::onViewWheelEvent(QWheelEvent* event)
     e.modifiers |= m & Qt::ControlModifier ? EVENTFLAG_CONTROL_DOWN : 0;
     e.modifiers |= m & Qt::ShiftModifier ? EVENTFLAG_SHIFT_DOWN : 0;
     e.modifiers |= m & Qt::AltModifier ? EVENTFLAG_ALT_DOWN : 0;
-    e.modifiers |= b & Qt::LeftButton ? EVENTFLAG_LEFT_MOUSE_BUTTON : 0;
```
Comment (tishion) [inline src/details/QCefViewPrivate.cpp:1039]: Why remove this?

File: src/details/QCefViewPrivate.cpp [hunk 2]
```diff
@@ -1036,18 +1036,22 @@ QCefViewPrivate::onViewWheelEvent(QWheelEvent* event)
...
     e.x = p.x();
     e.y = p.y();
 
-    // angleDelta().y() provides the angle through which the common vertical mouse wheel was rotated since the previous
-    // event. angleDelta().x() provides the angle through which the horizontal mouse wheel was rotated, if the mouse has
-    // a horizontal wheel; otherwise it stays at zero.
-    pCefBrowser_->GetHost()->SendMouseWheelEvent(
-      e, m & Qt::ShiftModifier ? d.x() : 0, m & Qt::ShiftModifier ? d.y() : d.y());
+    // Prevent diagonal scrolling: only allow scrolling in one direction at a time
+    if (d.x() != 0 && d.y() != 0) {
```
Comment (tishion) [inline src/details/QCefViewPrivate.cpp:1046]: Remove this condition check, it seems redundant

File: src/details/QCefViewPrivate.cpp [hunk 3]
```diff
@@ -1036,18 +1036,22 @@ QCefViewPrivate::onViewWheelEvent(QWheelEvent* event)
...
+      if (std::abs(d.x()) > std::abs(d.y())) {
```
Comment (tishion) [inline src/details/QCefViewPrivate.cpp:1047]: Can change to qAbs()?

#include <QtNumeric>

File: src/details/QCefViewPrivate.cpp (hunk 1 above)
Comment (EdnY1) [reply src/details/QCefViewPrivate.cpp:1039]: That was a mistake of mine when testing

**Comment by tishion** [comment]
I see what you are doing. Thanks for contribution. I am merging it now.

**Review by tishion — APPROVED** [review]

**Pull request merged**